refactor (#322)
KxSystems/docs: docs/kb/deferred-response.md

---
# Deferred response

## Overview

Ideally, for concurrency, all messaging would be async. However, sync messaging is a convenient paradigm for client apps. 

> Hence [`-30!x`](../basics/internal.md#-30x-deferred-response) was added to allow processing of a sync message to be ‘suspended’, by indicating the response for the currently-executing sync message will be sent explicitly later. This allows other messages to be processed prior to sending a response message. 

Reviewer: ```suggestion
You can use [`-30!x`](../basics/internal.md#-30x-deferred-response) to allow processing of a sync message to be ‘suspended’, by indicating the response for the currently-executing sync message to be sent explicitly later. This allows other messages to be processed prior to sending a response message. 
```
